rules/framework/vfs_overlay.bzl

```diff
@@ -62,15 +65,16 @@ def _build_subtrees(paths, vfs_prefix):
         # Loop the _framework_ path and add each dir to the current tree.
         # Assume the last bit is a file then add it as a file
         idx = 0
+
+        parts_len = len(parts)
         for part in parts:
-            if idx == len(parts) - 1:
-                ext_c = _get_external_contents(vfs_prefix, path_info.path)
+            if idx == parts_len - 1:
                 curr_subdirs.dict[part] = -1
-                curr_subdirs.json["contents"].append({"name": part, "type": "file", "external-contents": ext_c})
+                curr_subdirs.json["contents"].append({"name": part, "type": "file", "external-contents": vfs_prefix + path_info.path})
                 break
 
             # Lookup a value for the current subdirs, otherwise append
-            next_subdirs = curr_subdirs.dict.get(part, None)
+            next_subdirs = curr_subdirs.dict[part] if part in curr_subdirs.dict else None
```

Comment:
nit: think the `get` before was more clear

Reply:
this is also more performant 🤯, I don't remember by how much but `.get` seems to also cause more overhead than this other approach

Reply:
I think we only notice this in really big apps that call this many many times during the analysis phase

Reply:
Intersting

> Python has to do more work for dict.get():
>
> get is an attribute, so Python has to look this up, and then bind the descriptor found to the dictionary instance.
() is a call, so the current frame has to be pushed on the stack, a call has to be made, then the frame has to be popped again from the stack to continue.
The [...] notation, used with a dict, doesn't require a separate attribute step or frame push and pop.

https://stackoverflow.com/questions/36566331/why-does-dict-getkey-run-slower-than-dictkey